components/search/search.tsx

```diff
@@ -79,6 +82,30 @@ const Search: React.FC<Props> = (props: Props): JSX.Element => {
         new SearchUserProvider(actions.autocompleteUsersInTeam),
     ]);
 
+    const isDesktop = isDesktopApp() && isServerVersionGreaterThanOrEqualTo(getDesktopVersion(), '4.7.0');
+    if (enableFindShortcut) {
+        const handleKeyDown = useCallback((e: KeyboardEvent) => {
+            if (Utils.cmdOrCtrlPressed(e) && Utils.isKeyPressed(e, Constants.KeyCodes.F)) {
+                if (isDesktop || (!isDesktop && e.shiftKey)) {
+                    if (hideSearchBar) {
+                        actions.openRHSSearch();
+                        setKeepInputFocused(true);
+                    }
+                    handleUpdateSearchTerms(`in:${currentChannelName} `);
+                    handleFocus();
+                }
+            }
+        }, [currentChannelName, hideSearchBar]);
+
+        useEffect(() => {
+            document.addEventListener('keydown', handleKeyDown);
+
+            return () => {
+                document.removeEventListener('keydown', handleKeyDown);
+            };
+        }, [currentChannelName, hideSearchBar]);
```

Comment:
I am 3/5 that this needs to only be called once (like with `componentDidMount`), because every time `currentChannelName` or `hideSearchBar` changes it will add an additional event listener to the `document`

```suggestion
        }, []);
```

Reply:
The issue is that within the listener, `currentChannelName` and `hideSearchBar` are fixed when the listener is created. So if those values change (which they will depending on current channel and RHS state), the listener needs to be re-created.

@hmhealey recommended this pattern.

Reply:
Yeah, if I understand those hooks correctly, the props values get bound when the callback is created, so they wouldn't update without recreating the listener

Reply:
They are not really "bound" to the callback, but it certainly works a little like they would. 

useCallback creates a memoized instance of the function that's being passed in, so that it does not get re-defined on every small re-render.

Reply:
I mean I think that the values from props in the function are from when the callback is declared and that they're not updated as the props might change. This is as opposed to in a class component where `this.props.currentChannelName` always points to the latest version of the props.

That said, I like Caleb's version better where we just use a single `useEffect` and declare the callback inside of it. I wasn't a huge fan of using multiple hooks to begin with even though I might've proposed it